sdk/formrecognizer/ai-form-recognizer/src/lro/analyze.ts

```diff
@@ -132,6 +136,234 @@ export interface AnalyzeResult<Document = AnalyzedDocument> {
   documents: Document[];
 }
 
+/**
+ * A page within an analysis result.
+ */
+export interface DocumentPage {
+  /**
+   * 1-based page number in the input document.
+   */
+  pageNumber: number;
+
+  /**
+   * The general orientation of the content in clockwise direction, measured in degrees between (-180, 180].
+   */
+  angle: number;
+
+  /**
+   * The width of the image/PDF in pixels/inches, respectively.
+   */
+  width: number;
+
+  /**
+   * The height of the image/PDF in pixels/inches, respectively.
+   */
+  height: number;
+
+  /**
+   * The unit used by the width, height, and boundingBox properties. For images, the unit is "pixel". For PDF, the unit is "inch".
+   */
+  unit: LengthUnit;
+
+  /**
+   * Location of the page in the reading order concatenated content.
+   */
+  spans: DocumentSpan[];
+
+  /**
+   * Extracted words from the page.
+   */
+  words: DocumentWord[];
+
+  /**
+   * Extracted selection marks from the page.
+   */
+  selectionMarks?: DocumentSelectionMark[];
+
+  /**
+   * Extracted lines from the page, potentially containing both textual and visual elements.
+   */
+  lines: DocumentLine[];
+}
+
+/**
+ * Convert a REST-level DocumentPage into a convenience layer version.
+ * @param generated - a REST-level DocumentPage.
+ * @returns
+ */
+export function toDocumentPageFromGenerated(generated: GeneratedDocumentPage): DocumentPage {
+  // We will just overwrite the `lines` property with the transformed one rather than create a new object.
+  generated.lines = generated.lines.map((line) => toDocumentLineFromGenerated(line, generated));
+
+  return generated as DocumentPage;
+}
+
+/**
+ * A line of adjacent content elements on a page.
+ */
+export interface DocumentLine {
+  /**
+   * Concatenated content of the contained elements in reading order.
+   */
+  content: string;
+
+  /**
+   * Bounding box of the line.
+   */
+  boundingBox?: number[];
+
+  /**
+   * Location of the line in the reading order concatenated content.
+   */
+  spans: DocumentSpan[];
+
+  /**
+   * Compute the `DocumentWord`s that are related to this line.
+   *
+   * This function produces a lazy iterator that will yield one word before computing the next.
+   */
+  words: () => IterableIterator<DocumentWord>;
```

Comment:
This is how you initialize `words`:
```
(generated as DocumentLine).words = () => fastGetChildren(iterFrom(generated.spans, 0), page.words);
```

But I wonder is it really necessary to have `words` be a thunk when `fastGetChildren` is a generator? I mean the generator suspends its computation until `next()` gets called.

Reply:
I talked to @bterlson about this. The generator doesn't do any work until it is invoked via `next`, but unless this is a method, it cannot ever be "reset" or repeated. This also mimics other JavaScript constructs like `Map#keys()`

Reply:
I am not sure what do you mean. To clarify my point, consider the following example:
```ts
function* test(): IterableIterator<number> {
    yield 1;
    yield 2;
    yield 3;
}

for (const i of test()) {
    console.log(i)
}

for (const i of test()) {
    console.log(i)
}
```
This will print
```
1
2
3
1
2
3
```

Reply:
And that is how it is here:

```javascript
for (const word of line.words()) {
  // word is bound to a DocumentWord
}
```

If you are asking about why there is a function around the invocation of the generator (rather than just making `words` the generator function itself), it's because the generator needs the words from the related page as context. When it's invoked we need to provide that context, and I don't want to ask the user to provide it when we can always bind it correctly in the transform.

Reply:
ah I see it now, you're right.